cloudinit/net/eni.py

```diff
@@ -76,14 +76,22 @@ def _iface_add_subnet(iface, subnet):
         "dns_search",
         "dns_nameservers",
     ]
+    is_comp_addr = (
```

Comment:
Nit: This was a little hard to understand. I don't love nested defs, but I think I generally prefer them over lambdas. I also don't know what `comp` means...is it `complete`?

What about something like:
```python
    def addr_matches_ip_version(addr):
        return (":" in addr) == is_ipv6
```

Also, see my other comment about utility functions.